src/scene/Scale.js

```diff
@@ -187,8 +193,34 @@ function ordinal(scale, rng, group) {
     };
   }
 
-  prev.range = rng;
-  this._updated = true;
+  // Ordinal scale inversion requires range in ascending order.
+  // Shuffle domain/range to ensure this is true, and cache.
+  if (this._updated) {
```

Comment:
It looks like the primary strategy is to hold on to copies of the domain and range, but properly reversed if needed, in order to compute the inverse. Could we reduce data structure duplication here by instead maintaining a flag and then modifying the search procedure accordingly?